### DISCUSSION TITLE
Self-hosting solution for searching across multiple sites?

### DISCUSSION THREAD

**Discussion by huanlin** [google/docsy#2024] [Q&A] [answered]
Sorry if this quesion has been asked before.
I'm wondering is there a search tool for Hugo/Docsy that allows users to search across multiple sites (repositories)?
Or, is it even possible to achieve that?
My docs sites are behind a firewall, which means public services such as Google Search are not an option.

**Comment by fekete-robert** [comment]
Hi, you can host a full-blown search and indexing engine (like Apache Solr
or similar) and direct it to crawl your sites, and use its search page (or
integrate it into Docsy), but hosting and maintaining/managing such a site
is probably a lot of hassle.
Another option might be to use pagefind (https://pagefind.app/), which is
somewhat similar to lunr (so it's a client-side search engine). According
to its docs it can be used to search multiple sites, but I've never tried
that. Pagefind can be used with docsy (for example, we use it on
https://kube-logging.dev) with some work.

**Reply by huanlin** [reply]
Hi @fekete-robert , thank you for sharing. That is helpful!
Could you please shed more light on this? Cause in the Docsy documentation, it doesn't mention Pagefind, so I wonder is it easy to integrate Pagefind with Docsy?
I mean, is there any tutorials that you recommend, or do I just need to read the Pagefind documentation?

**Comment by fekete-robert** [comment, marked as answer]
Hi, I don't know any docsy-specific tutorials. Checking the docs is a good
starting point. The Logging operator docs that I've linked are open source
(and based on Docsy), so you can check the repo for an example
implementation (although it might be a bit overcomplicated. The main points
are:

- adding the pagefind scripts to layouts/partials/hooks/head-end.html
- setting up the search form in layouts/shortcodes/search-form.html and
including it in content/search.md
- generating the index files after the hugo build. For us that happens at
the end of the github workflow (see .github/workflows/publish-main.yaml)

**Reply by huanlin** [reply]
Thank you @fekete-robert , I'll take some time to try it. Much appreciated!
